Create 2022-04-04.md (#212)
solid/authentication-panel: meetings/2022-04-04.md

* AC: On the one hand, having PRs going to the spec repo can be useful for gating, but +1 on reviews happening in the editor's draft in tghe source repo. Change is easier to handle there, so it would be ideal. Realistically though, editors won't necessarily go to the source repos until a PR is opened on the solid-spec repo
* eP: Editors have editor's meeting, so we could signal the fact that new versions of the editor's draft is about to be published so that they can take action to review it. Reviews don't need to happen in a PR, but rather in the snapshot that is the draft once the PR has been merged.
* SC: +1. The notification panel right now branched the snapshots for inclusion to be included in the solid-specification repository. When a WG proposes a document be published, the reviewers review the snapshot to be added, and not the draft. Then the editor's draft is updated, and a new snapshot is created, and so on. No strong feeling about which repository should be reviewed. We don't want reviews to become unclear because it's unclear what reviewers are committing to.

> * AC: Right now we make a PR to the solid-spec, that's the snapshot. We can continue like this. There is an issue with automation though, because updating the draft to become a snapshot is very manual and error-prone. I've put together some automation to create a snapshot in a branch based on a tag. The challenge is with the first TR: there is currently nothing at the TR's URL. As soon as we merge something into this locaion, we can refer to it, but to be able to refer to it in the snapshot it needs to be created. If we merge the current snapshot in the solid-spec repo, a document will exist at the TR URL, but itself will refer to a non-existing document. Is it something all TR's face ?

Reviewer: ```suggestion
* AC: Right now we make a PR to the solid-spec; that's the snapshot. We can continue like this. There is an issue with automation though, because updating the draft to become a snapshot is very manual and error-prone. I've put together some automation to create a snapshot in a branch based on a tag. The challenge is that with the first TR, there is currently nothing at the TR's URL. As soon as we merge something into this location, we can refer to it, but to be able to refer to it in the snapshot, it needs to be created. If we merge the current snapshot in the solid-spec repo, a document will exist at the TR URL, but will itself refer to a non-existing document. Is this something all TRs face?
```

* eP: Snapshots are immutable, so updating them based on reviews doesn't really make sense. About the URLs, should they reflect the version numbers ?
* SC: 
  * The snapshot which is reviewed is the proposal to be published, and not the published document itself, which allows more flexibility for the URL issue.

>   * If the editor's draft isn't published yet, the TR URL can be omitted, or 404. 
>   * solidproject has a TR/ container, with resources intended to point at the latest snapshot of the sec. Each of these intermediary containers will then contain specific immutable versions of the iterative snapshots. The "latest" URL is the one we want to be passed around generally, and specific versions are more for implementations.

Reviewer: ```suggestion
  * If the editor's draft isn't published yet, the TR URL can be omitted, or `404`. 
  * solidproject has a `TR/` container, with resources intended to point at the latest snapshot of the spec. Each of these intermediary containers will then contain specific immutable versions of the iterative snapshots. The "latest" URL is the one we generally want to be passed around, and specific versions are more for implementations.
```

> * AC: Re versions in URLs, different groups do different things. VC use the same link for 1.0 and 1.1. RDF concepts has RDF-Concepts and RDF-Concepts/1.1, so basically we'll follow whatever the specification aligns on. It seems that a common pattern is to have a "latest" URL which will default to the latest versionned one, and then immutable versionned ones. 

Reviewer: ```suggestion
* AC: Re versions in URLs, different groups do different things. VC use the same link for 1.0 and 1.1. RDF concepts has RDF-Concepts and RDF-Concepts/1.1, so basically we'll follow whatever the specification aligns on. It seems that a common pattern is to have a "latest" URL which will default to the latest versioned one, and then immutable versioned ones. 
```

> * SC: We just want to be consistent with our persistence policy. The versioning policies are documented in the repo's CONTRIBUTING and/or README.

Reviewer: ```suggestion
* SC: We just want to be consistent with our persistence policy. The versioning policies are documented in the repo's `CONTRIBUTING` and/or `README`.
```

> * AC: Then we'll keep doing what we are doing, which seems aligned with what you're describing, and let us know if there's an issue.

Reviewer: ```suggestion
* AC: Then we'll keep doing what we are doing, which seems aligned with what you're describing. Let us know if there's an issue.
```

## Draft Status

> * AC: is this related to the previously discussed items ?
> * eP: The issue is that the status is used as a watermark in the default bikeshed style, which may be off-putting depending on the status we use. Should we override bikeshed's style, or which status should be used ?

Reviewer: ```suggestion
* AC: Is this related to the previously discussed items?
* eP: The issue is that the status is used as a watermark in the default bikeshed style, which may be off-putting depending on the status we use. Should we override bikeshed's style, or which status should be used?
```

* AC: The watermark is also an incentive to move into REC status 
* SC: As long as the document status is clear about endorsement and intellectual property, from a communication point of view it's okay to call the document an Editor's Draft or Unofficial Draft.

> * eP: We should also have the same consideration for the primer, it's typically a note.

Reviewer: ```suggestion
* eP: We should also have the same consideration for the primer; it's typically a note.
```

* SC: Since we're not using the W3C actual terminology in Solid, there isn't really any constraint on how we want to call the primer.
* AC: Bikeshed has a NOTE-WD for "Working draft" of a note, which works well for the primer
* eP: +1
* NAS: +1
* AC: And WG-NOTE for the final published version
* AC: I propose to iterate on the editor's draft until next week, and then open a new snapshot into the solid-spec repo, in order to keep a certain pace to get the Solid-OIDC spec moving.
* eP: +1

> * SC: Would the diff between the snapshots be just based on reviews, or would there be additional content ?

Reviewer: ```suggestion
* SC: Would the diff between the snapshots be just based on reviews, or would there be additional content?
```
